### PR TITLE
feat: add V-term with environment and workdir set

### PR DESCRIPTION
**Pull request by sfavazza**

### CODE REVIEW

**Comment by sfavazza** [comment]
@Silex can you have a look to this MR

File: docker-container.el [hunk 1]
```diff
@@ -234,7 +234,8 @@ nil, ask the user for it."
 ;;;###autoload (autoload 'docker-container-vterm "docker-container" nil t)
 (defun docker-container-vterm (container)
   "Open `vterm' in CONTAINER."
-  (interactive (list (docker-container-read-name)))
+  (interactive (list
+                (docker-container-read-name)))
   (if (fboundp 'vterm-other-window)
```
Comment (Silex) [inline docker-container.el:239]: Please remove this formatting

File: docker-container.el [hunk 2]
```diff
@@ -406,7 +435,8 @@ nil, ask the user for it."
    ("b" "Shell" docker-container-shell-selection)
    ("B" "Shell with env" docker-container-shell-env-selection)
    ("e" "Eshell" docker-container-eshell-selection)
-   ("v" "Vterm" docker-container-vterm-selection)])
+   ("v" "Vterm" docker-container-vterm-selection)
+   ("V" "Vterm with env"  docker-container-vterm-env-selection)])
```
Comment (Silex) [inline docker-container.el:440]: Please remove extra space (between `env"` and `docker-container`).

**Review by Silex — CHANGES_REQUESTED** [review]
Thanks, please check my comments.

**Review by Silex — APPROVED** [review]

**Comment by Silex** [comment]
Thanks 👍

**Comment by Silex** [comment]
Had to change from `aio-defun` to `defun` for it to work.

**Comment by sfavazza** [comment]
I don't understand why it does not work for you with `aio-defun`, now that has been changed to `defun` I got this error:

````elisp
Debugger entered--Lisp error: (error "‘iter-yield’ used outside a generator")
  signal(error ("‘iter-yield’ used outside a generator"))
  error("`iter-yield' used outside a generator")
  iter-yield(#s(aio-promise nil nil))
  docker-container-vterm-env("odindevenv-tftp_srv-1")
  #f(compiled-function () (interactive nil) #<bytecode 0x164259ed9919ee3f>)()
  apply(#f(compiled-function () (interactive nil) #<bytecode 0x164259ed9919ee3f>) nil)
  #f(compiled-function (fn &rest args) #<bytecode 0xe55b325266e3da8>)(#f(compiled-function () (interactive nil) #<bytecode 0x164259ed9919ee3f>))
  apply(#f(compiled-function (fn &rest args) #<bytecode 0xe55b325266e3da8>) #f(compiled-function () (interactive nil) #<bytecode 0x164259ed9919ee3f>) nil)
  (lambda (fn &rest args) (interactive #f(compiled-function (spec) #<bytecode -0x10bbfb860b322f72>)) (apply '#f(compiled-function (fn &rest args) #<bytecode 0xe55b325266e3da8>) fn args))(#f(compiled-function () (interactive nil) #<bytecode 0x164259ed9919ee3f>))
  apply((lambda (fn &rest args) (interactive #f(compiled-function (spec) #<bytecode -0x10bbfb860b322f72>)) (apply '#f(compiled-function (fn &rest args) #<bytecode 0xe55b325266e3da8>) fn args)) #f(compiled-function () (interactive nil) #<bytecode 0x164259ed9919ee3f>) nil)
  docker-container-vterm-env-selection()
  funcall-interactively(docker-container-vterm-env-selection)
  call-interactively(docker-container-vterm-env-selection nil nil)
  command-execute(docker-container-vterm-env-selection)
```

This is due to the call to `aio-await` in the `docker-container-vterm-env`. It is the same approach like in the `docker-container-shell-env`. Can you please revert your change?

**Comment by Silex** [comment]
Hum, right I should have also replaced the `aio-await` with a sync version.

Anyway I reverted my change, but please help me debug this then: https://github.com/Silex/docker.el/actions/runs/6101476986

**Comment by Silex** [comment]
@sfavazza

**Comment by sfavazza** [comment]
I don't understand why the `vterm-other-window` function is not found. The fboundp test is performed also in the `docker-container-vterm` but it does not give any problem. I am investigating, but it does not seems trivial. Do you have any idea whether the `aio` macros may have an impact on the loading order?

image

**Comment by Silex** [comment]
@LaurenceWarne: I summon you here because you might have an idea why we have these errors.

https://github.com/skeeto/emacs-aio is a bit special so there is maybe something we are missing.

**Comment by LaurenceWarne** [comment]
Experimentally, the `fboundp` call appears to have to directly precede the statement.  It would be interesting to know more about it, though I can't appear to find any doc?

Opened a PR anyway: https://github.com/Silex/docker.el/pull/223.

**Comment by sfavazza** [comment]
@LaurenceWarne much appreciated your intervention

**Pull request merged**